templates/ingress.yaml

```diff
@@ -370,86 +370,126 @@ metadata:
   {{- end }}
 spec:
 {{- include "coder.ingress.tls" . }}
-  backend:
-    serviceName: {{ include "coder.serviceName" . }}
-    servicePort: 8080
+  defaultBackend:
+    service:
+      name: {{ include "coder.serviceName" . }}
+      port:
+        number: 8080
   {{- if not .Values.coderd.replica.enable }}
   rules:
   - host: {{ .Values.ingress.host | quote }}
     http:
       paths:
     {{- if or (not .Values.ingress.usePathWildcards) .Values.ingress.useDefault }}
       {{- if hasKey .Values "envproxy" }}
-      - path: /proxy/
+      - path: /proxy
+        pathType: Prefix
         backend:
-          serviceName: envproxy
-          servicePort: 8080
+          service:
+            name: envproxy
+            port:
+              number: 8080
       {{- end }}
-      - path: /api/
+      - path: /api
+        pathType: Prefix
         backend:
-          serviceName: {{ include "coder.serviceName" . }}
-          servicePort: 8080
-      - path: /auth/
+          service:
+            name: {{ include "coder.serviceName" . }}
+            port:
+              number: 8080
+      - path: /auth
+        pathType: Prefix
         backend:
-          serviceName: {{ include "coder.serviceName" . }}
-          servicePort: 8080
+          service:
+            name: {{ include "coder.serviceName" . }}
+            port:
+              number: 8080
       - path: /
+        pathType: Prefix
         backend:
           {{- if hasKey .Values "dashboard" }}
-          serviceName: dashboard
-          servicePort: 3000
+          service:
+            name: dashboard
+            port:
+              number: 3000
           {{- else }}
-          serviceName: {{ include "coder.serviceName" . }}
-          servicePort: 8080
+          service:
+            name: {{ include "coder.serviceName" . }}
+            port:
+              number: 8080
           {{- end }}
     {{- if ne .Values.devurls.host "" }}
   - host: {{ .Values.devurls.host | quote }}
     http:
       paths:
       - path: /
+        pathType: Prefix
         backend:
           {{- if hasKey .Values "envproxy" }}
-          serviceName: envproxy
-          servicePort: 8080
+          service:
+            name: envproxy
+            port:
+              number: 8080
           {{- else }}
-          serviceName: {{ include "coder.serviceName" . }}
-          servicePort: 8080
+          service:
+            name: {{ include "coder.serviceName" . }}
+            port:
+              number: 8080
           {{- end }}
     {{- end }}
     {{- else }}
-      - path: /proxy/*
+      - path: /proxy
```

Comment:
This seems to obviate the `usePathWildcards` option. Do we no longer need them? Last time I played with this, `gce-ingress` required wildcards, but maybe this is fixed now by specifying `pathType: Prefix`. If so we should just remove that option.

Reply:
I did test with my local dev deploy but am not sure how to test otherwise since I'm not familiar with what the problem was

Reply:
I'll test it out

Reply:
@coadler What was the outcome of your testing? Should we remove that option?